### ISSUE TITLE
Word wrap?

### ISSUE THREAD

**Issue by chriskrycho** [onivim/oni2#1444]
I may have just missed this, but it seems there isn't yet support for soft word wrapping at the editor width or other boundaries?

**Comment by glennsl** [comment]
I don't think you've missed anything. `:set textwidth` does not seem to do a thing.

**Comment by Spirarel** [comment]
To add to this, there is no horizontal scroll bars, so you can only see text if you resize the window.

Reproducible on a fresh 0.4.0 when opening keybindings.json
macOS 10.14.6

**Comment by mseri** [comment]
I am hitting the same issue, for the moment this is a blocker for me

**Comment by jerabaul29** [comment]
Any update? This is a big blocker for me.

**Comment by mainderay** [comment]
Seems to be working for me in the latest release!

**Comment by jerabaul29** [comment]
Ok, how did you get it to wrap @mainderay ? Tried yesterday with the latest release but no success.

**Comment by mainderay** [comment]
Add these lines to the configuration.json file:

  "experimental.viml": [
    "set wrap", "set textwidth=60"
  ]

I had them in there for months with no effect and suddenly wrap started working!

-r

**Comment by mseri** [comment]
Thanks for sharing! Unfortunately it is sill not working for me on osx

**Comment by mainderay** [comment]
Hmmm. I’m on OSX as well. I will say that it only breaks on white space at the moment, so if one types a long continuous string, no break will occur. Does that help? :-)

**Comment by mseri** [comment]
Oh, I see what happened. Indeed, the hard wrap works now using your configuration.
I was looking at the soft wrap when I tested it, and unfortunately
```
"experimental.viml": [
    "set wrap", "set linebreak", "set nolist",
    "set textwidth=0", "set wrapmargin=0"
]
```
is not working yet.

The hard wrap working is already quite helpful though, once also the soft one works I will be able to fully switch.

**Comment by tkolleh** [comment]
Any updates on enable / disabling word wrapping by default?

**Comment by mseri** [comment]
Both hard wrap as mentioned above and soft wrap by setting `"experimental.editor.wordWrap": "on"` should work now. I don’t use the hard wrap, but soft wrap is working very nicely for me!

**Comment by tkolleh** [comment]
That works, thanks. How can a newcomer like myself find that answer without reaching out to the community. The teach them how to fish approach. Are there docs that I can explore within (or outside of) onivim or is this something that best left to CTRL-F <some related word or phrase>?

**Comment by bryphe** [comment]
> I don’t use the hard wrap, but soft wrap is working very nicely for me!

Awesome! Thanks for testing this out @mseri :+1: 

I've been using it as well - I think it's a good time to promote it out of experimental - I just opened a PR for it: https://github.com/onivim/oni2/pull/2783 (there are still extra features that would be nice, like being able to specify a wrap column - but that can come later). Just a heads up - the configuration will need to be updated to `editor.wordWrap` from `experimental.editor.wordWrap`. It'd be nice to have a more graceful path (allowing for 'deprecated' configuration settings).

> How can a newcomer like myself find that answer without reaching out to the community. The teach them how to fish approach.

Good question @tkolleh  - we do have these docs here: https://onivim.github.io/docs/configuration/settings (they do need improving though, any feedback you have is welcome). I often skip documenting experimental features until they've baked a bit - but that will be fixed for this particular case in #2783
